### ISSUE TITLE
torch matrix multiplication shape disagreement

### ISSUE THREAD

**Issue by simin75simin** [IrvingMeng/MagFace#35]
so i had to change line 144 of models/iresnet.py to x = x.view(x.size(0)*4, -1) in order to get it running. this probably has to do with torch updates. here are my stats:
os: ubuntu 18.04
nvcc 9.1
cuda: 11.4
python 3.6.9
pip freeze:
aiocontextvars==0.2.2
anyio==3.4.0
argon2-cffi==21.3.0
argon2-cffi-bindings==21.2.0
async-generator==1.10
attrs==21.2.0
Babel==2.9.1
backcall==0.2.0
bleach==4.1.0
certifi==2021.10.8
cffi==1.15.0
charset-normalizer==2.0.9
contextvars==2.4
cycler==0.11.0
dataclasses==0.8
decorator==4.4.2
defusedxml==0.7.1
entrypoints==0.3
idna==3.3
imageio==2.9.0
immutables==0.16
importlib-metadata==4.8.2
ipykernel==5.5.6
ipython==7.16.2
ipython-genutils==0.2.0
jedi==0.17.2
Jinja2==3.0.3
joblib==1.1.0
json5==0.9.6
jsonschema==4.0.0
jupyter-client==7.1.0
jupyter-core==4.9.1
jupyter-server==1.13.1
jupyterlab==3.2.5
jupyterlab-pygments==0.1.2
jupyterlab-server==2.9.0
kiwisolver==1.3.1
loguru==0.5.3
MarkupSafe==2.0.1
matplotlib==3.3.4
mistune==0.8.4
nbclassic==0.3.4
nbclient==0.5.9
nbconvert==6.0.7
nbformat==5.1.3
nest-asyncio==1.5.4
networkx==2.5.1
notebook==6.4.6
numpy==1.19.5
opencv-contrib-python==4.5.4.60
packaging==21.3
pandas==1.1.5
pandocfilters==1.5.0
parso==0.7.1
pexpect==4.8.0
pickleshare==0.7.5
Pillow==6.2.2
pkg_resources==0.0.0
prometheus-client==0.12.0
prompt-toolkit==3.0.24
ptyprocess==0.7.0
pycparser==2.21
Pygments==2.10.0
pyparsing==3.0.6
pyrsistent==0.18.0
python-dateutil==2.8.2
pytz==2021.3
PyWavelets==1.1.1
pyzmq==22.3.0
requests==2.26.0
scikit-image==0.17.2
scikit-learn==0.24.2
scipy==1.5.4
seaborn==0.11.2
Send2Trash==1.8.0
six==1.16.0
sklearn==0.0
sniffio==1.2.0
termcolor==1.1.0
terminado==0.12.1
testpath==0.5.0
threadpoolctl==3.0.0
tifffile==2020.9.3
torch==1.10.0
torchshard==0.1.0
torchvision==0.11.1
tornado==6.1
tqdm==4.62.3
traitlets==4.3.3
typing_extensions==4.0.1
urllib3==1.26.7
wcwidth==0.2.5
webencodings==0.5.1
websocket-client==1.2.3
zipp==3.6.0
